build/utils/dotnet/solution/solution.go

```diff
@@ -174,14 +174,27 @@ func (solution *solution) loadProjects(slnProjects []project.Project, log utils.
 	return nil
 }
 
-func (solution *solution) parseProjectsFromSolutionFile(slnProjects []string, log utils.Log) ([]project.Project, error) {
+func (solution *solution) parseProjectsFromSolutionFile(slnProjects []string, excludePattern string, log utils.Log) ([]project.Project, error) {
 	var projects []project.Project
 	for _, projectLine := range slnProjects {
 		projectName, projFilePath, err := parseProjectLine(projectLine, solution.path)
 		if err != nil {
 			log.Error(err)
 			continue
 		}
+		// Exclude projects by pattern.
+		if len(excludePattern) > 0 {
```

Comment:
consider putting this logic into a function like "isPatternExcluded) or something for the clarity of the flow